src/Race/Race.ts

```diff
@@ -100,7 +101,14 @@ export default class Race {
   }
 
   async addParticipant(interaction: ButtonInteraction) {
-    const participant = interaction.user;
+    const { member, user } = interaction
+    const participant = user as DiscracedUser;
+    participant.renderName = user.username;
+
+    if (member instanceof GuildMember && member.nickname) {
```

Comment:
You're taking care of it already, but JC tipped me off that [GuildMember.displayName](https://discord.js.org/#/docs/discord.js/stable/class/GuildMember?scrollTo=displayName) takes care of the fallback for you.